modin/core/dataframe/pandas/partitioning/axis_partition.py

```diff
@@ -134,6 +134,88 @@ def apply(
             )
         )
 
+    def split(
+        self, split_func, num_splits, f_args=None, f_kwargs=None, extract_metadata=False
+    ):
+        """
+        Split axis partition into multiple partitions using the `split_func`.
+
+        Parameters
+        ----------
+        split_func : callable(pandas.DataFrame) -> list[pandas.DataFrame]
+            A function that takes partition's content and split it into multiple chunks.
+        num_splits : int
+            The number of splits the `split_func` return.
+        f_args : iterable, optional
+            Positional arguments to pass to the `split_func`.
+        f_kwargs : dict, optional
+            Keyword arguments to pass to the `split_func`.
+        extract_metadata : bool, default: False
```

Comment:
The original `.split` method is already implemented in the manner of not extracting the metadata:
https://github.com/modin-project/modin/blob/cd7611cd6d7fd4054e06017d879234261293561b/modin/core/dataframe/pandas/partitioning/partition.py#L394-L398
So the full-axis implementation just follows the initial approach.

We don't want to extract metadata because:
1. Partitions generated by this function are temporary, at the reshuffling flow the `split_row_partitions` are immediately replaced by `new_partitions` holding new metadata, meaning that the metadata of `split_row_partitions` is never accessed:
   https://github.com/modin-project/modin/blob/cd7611cd6d7fd4054e06017d879234261293561b/modin/core/dataframe/pandas/partitioning/partition_manager.py#L1600-L1608
2. The splitting stage generates a lot of partitions (up to `ncores ^ 2`), it's already not an easy task for ray to put into storage that big amount of futures at once, the situation becomes even worse when we ask to store the metadata futures as well (`4 * (ncores ^ 2)` amount of futures at once). I've measured the case from #5533 with and without the partition's metadata, and received a stable 9% speed-up (~ 0.12s) for the case without metadata.